File: solutions/r/grains/1/grains.R
square <- function(n) {
  if (!n %in% 1:64) stop("Invalid value: please choose between 1 and 64")
  
  2^(n-1)
}

total <- function() {
  # based on formula for sum of the first n terms of a geometric series (rearranged)
  -(1-2^64) 
}

File: solutions/typescript/grains/1/grains.ts
export const square = (n: number): BigInt => {
  if (n < 1 || n > 64) throw new Error('Argument must be between 1 and 64.')

  return BigInt(2 ** (n - 1))
}

export const total = () => {
  return BigInt(2 ** 64) - BigInt(1)
}
